optuna/integration/skopt.py

```diff
@@ -0,0 +1,255 @@
+from __future__ import absolute_import
+
+import numpy as np
+
+import optuna
+from optuna import distributions
+from optuna import samplers
+from optuna.samplers import BaseSampler
+from optuna import structs
+from optuna.structs import StudyDirection
+from optuna import types
+
+try:
+    import skopt
+    from skopt.space import space
+
+    _available = True
+except ImportError as e:
+    _import_error = e
+    # SkoptSampler is disabled because Scikit-Optimize is not available.
+    _available = False
+
+if types.TYPE_CHECKING:
+    from typing import Any  # NOQA
+    from typing import Dict  # NOQA
+    from typing import List  # NOQA
+    from typing import Optional  # NOQA
+    from typing import Tuple  # NOQA
+
+    from optuna.distributions import BaseDistribution  # NOQA
+    from optuna.structs import FrozenTrial  # NOQA
+    from optuna.study import InTrialStudy  # NOQA
+
+
+class SkoptSampler(BaseSampler):
+    """Sampler using Scikit-Optimize as the backend.
+
+    Example:
+
+        Optimize a simple quadratic function by using :class:`~optuna.integration.SkoptSampler`.
+
+        .. code::
+
+                def objective(trial):
+                    x = trial.suggest_uniform('x', -100, 100)
+                    y = trial.suggest_int('y', -10, 10)
+                    return x**2 + y
```

Comment:
I think the search space seems too wide compared with the number of trials.
When I executed the current code three times, the average of the best values is about 10. The error seems big, and users may think that the search efficiency of SkoptSampler is not so good.

What do you think of using a narrower search space?
For instance, the following setting marked `2.1e-9` as the average of the best score.

```python
def objective(trial):
    x = trial.suggest_uniform('x', -10, 10)
    y = trial.suggest_int('y', 0, 10)
    return x**2 + y
```

Reply:
Sounds good. Thank you for your suggestion.

Fixed: https://github.com/pfnet/optuna/pull/399/commits/fe3279e9447254e71ea5f8edf25e18af9044aca1